app/src/main/java/com/hedvig/app/feature/embark/passages/previousinsurer/retrieveprice/StartDataCollectionUseCase.kt

```diff
@@ -0,0 +1,101 @@
+package com.hedvig.app.feature.embark.passages.previousinsurer.retrieveprice
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.ExternalInsuranceProviderV2Query
+import com.hedvig.android.owldroid.graphql.InitiateDataCollectionNOMutation
+import com.hedvig.android.owldroid.graphql.InitiateDataCollectionSEMutation
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+import java.lang.IllegalArgumentException
+import java.util.UUID
+
+interface StartDataCollectionUseCase {
+    suspend fun startDataCollection(
+        personalNumber: String,
+        insuranceProvider: String,
+    ): DataCollectionResult
+}
+
+class StartDataCollectionUseCaseImpl(
+    val apolloClient: ApolloClient,
+    val marketManager: MarketManager
+) : StartDataCollectionUseCase {
+
+    override suspend fun startDataCollection(
+        personalNumber: String,
+        insuranceProvider: String,
+    ): DataCollectionResult {
+        val reference = UUID.randomUUID().toString()
+
+        val mutation = createMutation(
```

Comment:
Maybe a bit more descriptive name here since we have more than 1 mutation/query happening in this file?